### PR TITLE
T1 26.register

### PR DESCRIPTION
**Pull request by Cortesjohnj**
# User Registration via backend
## key points

- I've created user registration via backend (foundations or users) by sending the required information established in the User Schema and saving it at mongo db.
- It validates if an other mail is already taken.
## Aditional features

- I've added and aditional controller to ger the information of a user via email.

User Register ok 1
User Register ok 2
User Register error 2
User Register error 1
foundation Register ok 2
foundation Register ok 1
foundation Register error 2
foundation Register error 1

### CODE REVIEW

File: controllers/controllers.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const User = require("../models/User");
+const Foundation = require("../models/Foundation");
+
+const createUser = async (req, res, next) => {
+  console.log(req.body.role);
+  if (req.body.role === "user") {
```
Comment (leoneloliveros) [inline controllers/controllers.js:6]: This two conditions are almost the same, you can use one try - catch and add conditions in try 
calling their respective model, something like this: 

`try {
      let newUser;
      if (role === 'user') {
        newUser = await new User(req.body)
      } else {
        newUser = await new Foundation(req.body)
      }
      await newUser.save()
      res.status(201).json(newUser)
} catch (err) {
      console.log("Validation Error:", err.errors);
      res.status(422).json(err.errors);
}`

File: middlewares/middelwares.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const User = require("../models/User");
+const jwt = require("jsonwebtoken");
+
+const auth = async (req, res, next) => {
+  try {
+    const token = req.get("Authorization");
+    const data = jwt.verify(token, "secret key"); // { userId: "12345"}
+    console.log("Data del token:", data);
+    console.log("UserId: ", data.userId);
```
Comment (leoneloliveros) [inline middlewares/middelwares.js:9]: Please remove console.logs

File: controllers/controllers.js [hunk 3]
```diff
@@ -4,6 +4,30 @@ const jwt = require("jsonwebtoken");
 const config = require("../config/index");
 const Pet = require("../models/Pet");
 
+const createUser = async (req, res, next) => {
+  try {
+    let newUser;
+    console.log(req.body);
+    if (req.body.role === "user" || req.body === "admin") {
+      newUser = await new User(req.body);
+      await newUser.save();
```
Comment (leoneloliveros) [inline controllers/controllers.js:13]: You can use outside of this two conditions, 
`
if () { }
else { }

await newUser.save()
res.status(201).json(newUser);
`

**Review by leoneloliveros — APPROVED** [review]
Approved! Pending refactor in createUser and later please marge this branch

File: controllers/controllers.js (hunk 3 above)
Comment (Cortesjohnj) [reply controllers/controllers.js]: Thanks, refactored code!

**Pull request merged**